### ISSUE TITLE
behaviour of allow_duplicates unclear

### ISSUE THREAD

**Issue by philfry** [ansible/ansible#43199]
##### SUMMARY
`allow_duplicates` is behaving strangely when mixing methods to include a role (`roles` in the playbook, `dependencies` in meta, `include_role` in tasks).

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
core

##### ANSIBLE VERSION
```
ansible 2.6.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/xxx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
```

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE
```bash
install -D <(echo -e "---\ndependencies: [ role: role2 ]") roles/role1/meta/main.yml
install -D <(echo -e "---\nallow_duplicates: false") roles/role2/meta/main.yml
install -D <(echo -e "---\n- debug: msg='running {{role_name}}'\n- include_role: name=role2 allow_duplicates=false") roles/role1/tasks/main.yml
install -D <(echo -e "---\n- set_fact: times={{times|int+1}}\n- debug: msg='running {{role_name}} x{{times}}'") roles/role2/tasks/main.yml
cat > playbook.yml <<EOF
---
- hosts: localhost
  gather_facts: no
  vars:
    times: 0
  roles:
    - role1
  tasks:
    - debug: msg="marker 1"
    - include_role: name=role2 allow_duplicates=false
    - debug: msg="marker 2"
    - include_role: name=role2 allow_duplicates=false
    - debug: msg="marker 3"
    - include_role: name=role1 allow_duplicates=false
EOF
```

##### EXPECTED RESULTS
`role2` running only once.

##### ACTUAL RESULTS
`role2` running five times.

```
PLAY [localhost] ***************************************************************

TASK [role2 : set_fact] ********************************************************
ok: [localhost]

TASK [role2 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role2 x1"
}

TASK [role1 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role1"
}

TASK [include_role : role2] ****************************************************

TASK [role2 : set_fact] ********************************************************
ok: [localhost]

TASK [role2 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role2 x2"
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "marker 1"
}

TASK [include_role : role2] ****************************************************

TASK [role2 : set_fact] ********************************************************
ok: [localhost]

TASK [role2 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role2 x3"
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "marker 2"
}

TASK [include_role : role2] ****************************************************

TASK [role2 : set_fact] ********************************************************
ok: [localhost]

TASK [role2 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role2 x4"
}

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "marker 3"
}

TASK [include_role : role1] ****************************************************

TASK [role1 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role1"
}

TASK [include_role : role2] ****************************************************

TASK [role2 : set_fact] ********************************************************
ok: [localhost]

TASK [role2 : debug] ***********************************************************
ok: [localhost] => {
    "msg": "running role2 x5"
}

PLAY RECAP *********************************************************************
localhost                  : ok=15   changed=0    unreachable=0    failed=0   
```

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/64121
